app/routes/crate/version-dependencies.js

```diff
@@ -2,21 +2,23 @@ import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
 
 export default class VersionRoute extends Route {
+  @service store;
   @service router;
 
   async model(params, transition) {
     let crate = this.modelFor('crate');
 
-    let versions;
-    try {
-      versions = await crate.loadVersionsTask.perform();
-    } catch (error) {
-      let title = `${crate.name}: Failed to load version data`;
-      return this.router.replaceWith('catch-all', { transition, error, title, tryAgain: true });
-    }
-
     let requestedVersion = params.version_num;
-    let version = versions.find(version => version.num === requestedVersion);
+    let version =
+      crate.loadedVersionsByNum.get(requestedVersion) ??
+      (await this.store
+        .queryRecord('version', {
+          name: crate.id,
+          num: requestedVersion,
+        })
+        .catch(() => {
+          // ignored
+        }));
```

Comment:
AFAIU this change removes proper error handling. if the version could not be loaded because of a random network issue if would now show "Version not found" instead of the "Failed to load version data" with a retry button that was removed above